src/import/load-costume.js

```diff
@@ -25,55 +25,54 @@ const loadCostume = function (md5ext, costume, runtime) {
-        return costumeAsset;
-    });
-
     if (!runtime.renderer) {
         log.error('No rendering module present; cannot load costume asset: ', md5ext);
-        return promise.then(() => costume);
+        return costume;
+    }
+    const AssetType = runtime.storage.AssetType;
+    costume.assetId = costumeAsset.assetId;
+    if (costumeAsset.assetType === AssetType.ImageVector) {
+        costume.skinId = runtime.renderer.createSVGSkin(costumeAsset.decodeText(), rotationCenter);
+        return costume;
     }
 
-    if (assetType === AssetType.ImageVector) {
-        promise = promise.then(costumeAsset => {
-            costume.skinId = runtime.renderer.createSVGSkin(costumeAsset.decodeText(), rotationCenter);
-            return costume;
-        });
-    } else {
-        promise = promise.then(costumeAsset => (
-            new Promise((resolve, reject) => {
-                const imageElement = new Image();
-                const onError = function () {
+    return new Promise((resolve, reject) => {
+        const imageElement = new Image();
+        const onError = function () {
                     // eslint-disable-next-line no-use-before-define
-                    removeEventListeners();
-                    reject();
-                };
-                const onLoad = function () {
+            removeEventListeners();
+            reject();
+        };
+        const onLoad = function () {
                     // eslint-disable-next-line no-use-before-define
-                    removeEventListeners();
-                    resolve(imageElement);
-                };
-                const removeEventListeners = function () {
-                    imageElement.removeEventListener('error', onError);
-                    imageElement.removeEventListener('load', onLoad);
-                };
-                imageElement.addEventListener('error', onError);
-                imageElement.addEventListener('load', onLoad);
-                imageElement.src = costumeAsset.encodeDataURI();
-            })
-        )).then(imageElement => {
-            costume.skinId = runtime.renderer.createBitmapSkin(imageElement, costume.bitmapResolution, rotationCenter);
-            return costume;
-        });
-    }
-    return promise;
+            removeEventListeners();
+            resolve(imageElement);
+        };
+        const removeEventListeners = function () {
+            imageElement.removeEventListener('error', onError);
+            imageElement.removeEventListener('load', onLoad);
+        };
+        imageElement.addEventListener('error', onError);
+        imageElement.addEventListener('load', onLoad);
+        imageElement.src = costumeAsset.encodeDataURI();
+    }).then(imageElement => {
+        costume.skinId = runtime.renderer.createBitmapSkin(imageElement, costume.bitmapResolution, rotationCenter);
+        return costume;
+    });
 };
 
+loadCostume.loadCostumeFromAsset = loadCostumeFromAsset;
```

Comment:
Hmm... putting a function on a non-constructor function feels a bit out-of-convention to me. Maybe we should make a `LoadCostume` object whose job is to contain the various (possibly static) methods used in loading costumes? Alternatively, maybe `module.exports = {loadCostume, loadCostumeFromAsset};`? (Kinda the same thing, but maybe more lightweight...)